sycl/include/CL/sycl/detail/cg_types.hpp

```diff
@@ -179,31 +179,29 @@ struct KernelLambdaHasKernelHandlerArgT {
 // Helpers for running kernel lambda on the host device
 
 template <typename KernelType>
-typename detail::enable_if_t<
-    KernelLambdaHasKernelHandlerArgT<KernelType>::value, void>
+typename std::enable_if_t<KernelLambdaHasKernelHandlerArgT<KernelType>::value>
```

Comment:
I believe the keyword 'typename' is not required when you use std::enable_if_t, but Ok. Thank you for the additional changes anyway.